src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -342,46 +320,59 @@ internal static bool TryValidateParseStyleInteger(NumberStyles style, [NotNullWh
             return true;
         }
 
-        internal static ParsingStatus TryParseBigInteger(string? value, NumberStyles style, NumberFormatInfo info, out BigInteger result)
+        internal static unsafe ParsingStatus TryParseBigInteger(ReadOnlySpan<char> value, NumberStyles style, NumberFormatInfo info, out BigInteger result)
         {
-            if (value == null)
+            if (!TryValidateParseStyleInteger(style, out ArgumentException? e))
             {
-                result = default;
-                return ParsingStatus.Failed;
+                throw e; // TryParse still throws ArgumentException on invalid NumberStyles
             }
 
-            return TryParseBigInteger(value.AsSpan(), style, info, out result);
+            if ((style & NumberStyles.AllowHexSpecifier) != 0)
+            {
+                return TryParseBigIntegerHexNumberStyle(value, style, out result);
+            }
+
+            return TryParseBigIntegerNumber(value, style, info, out result);
         }
 
-        internal static ParsingStatus TryParseBigInteger(ReadOnlySpan<char> value, NumberStyles style, NumberFormatInfo info, out BigInteger result)
+        internal static unsafe ParsingStatus TryParseBigIntegerNumber(ReadOnlySpan<char> value, NumberStyles style, NumberFormatInfo info, out BigInteger result)
         {
-            if (!TryValidateParseStyleInteger(style, out ArgumentException? e))
+
+            scoped Span<byte> buffer;
+            byte[]? arrayFromPool = null;
+
+            if (value.Length < 255)
             {
-                throw e; // TryParse still throws ArgumentException on invalid NumberStyles
+                buffer = stackalloc byte[value.Length + 1 + 1];
             }
-
-            BigNumberBuffer bigNumber = BigNumberBuffer.Create();
-            if (!FormatProvider.TryStringToBigInteger(value, style, info, bigNumber.digits, out bigNumber.precision, out bigNumber.scale, out bigNumber.sign))
+            else
             {
-                result = default;
-                return ParsingStatus.Failed;
+                buffer = arrayFromPool = ArrayPool<byte>.Shared.Rent(value.Length + 1 + 1);
```

Comment:
it's the second time 1 + 1 is being used and as a person who reads the code for the first time it's not obvious to me. Could you please add a comment? Or introduce a const with-self-describing-name and use it in both places?